### DISCUSSION TITLE
Lag occuring when dragging on large ER diagram

### DISCUSSION THREAD

**Discussion by jibinjoy8897** [clientIO/joint#2527] [Q&A]
There seems to be a lag while dragging nodes in a large diagram.The diagram has 191 table nodes which are dynamically created from database tables.Small diagram having node count less than 100 doesn’t have this lag and is very smooth while dragging.Is there any way to solve this dragging lag?

**Comment by kumilingus** [comment]
- you can employ virtual rendering: https://www.jointjs.com/demos?feature=Virtual+rendering
- you can implement dragging preview instead of immediate dragging: https://www.jointjs.com/demos/data-mapping
- for large graphs is best to avoid using heavy routers and connectors (such as `manhattan`, `metro` or `jumpover`).
